src/Lime.Client.TestConsole/ViewModels/EnvelopeViewModel.cs

```diff
@@ -151,5 +153,52 @@ public bool IsRaw
         }
 
         #endregion Data Properties
+
+        public static EnvelopeViewModel Parse(string inputJson)
+        {
+            try
+            {
+                var jsonObject = JObject.Parse(inputJson);
+
+                if (jsonObject.HasValues)
+                {
+                    var envelopeViewModel = new EnvelopeViewModel
+                    {
+                        Json = inputJson
+                    };
+
+                    if (envelopeViewModel.Envelope != null)
+                    {
+                        return envelopeViewModel;
+                    }
+                    else
+                    {
+                        throw new ArgumentException("The input is a valid JSON document, but is not an Envelope");
+                    }
+                }
+                else
+                {
+                    throw new ArgumentException("The input is a invalid or empty JSON document");
+                }
+            }
+            catch
```

Comment:
Here this simple catch is swallowing the more specific `ArgumentException`'s above.
You should add a condition exception here, for instance:
```catch (Exception e) when (!(e is ArgumentException))```

BTW, whem throwing a new Exception from another, is good to add it as its inner exception:
```throw new ArgumentException("The input is a invalid JSON document", e);```